persist/sqlite/consensus.go

```diff
@@ -1204,3 +1204,17 @@ func (s *Store) UpdateChainState(reverted []chain.RevertUpdate, applied []chain.
 		return nil
 	})
 }
+
+// ResetChainState implements explorer.Store
+func (s *Store) ResetChainState() error {
+	return s.transaction(func(tx *txn) error {
+		tables := []string{"blocks", "transactions", "v2_transactions", "host_info", "address_balance", "state_tree"}
```

Comment:
I'd spin up a mainnet explorer, let it sync and then manually run this query on these tables. Then check the database and see if it's in the state you'd expect. To make sure it works on a mature database.

Reply:
I tried on nearly fully synced mainnet but it was taking so long that I cancelled before it finished.  In that case we might just have to drop all the tables and recreate them.

Reply:
Ok tested with new logic performance is good now